### ISSUE TITLE
Add possibility to add colorbar to contour overlay

### ISSUE THREAD

**Issue by JohannesWiesner** [nilearn/nilearn#2136]
It would be nice if one could also add a color bar when using `nilearn.plotting.displays.OrthoSlicer.add_contours`. Is there a reason why the visualization of a color bar is implemented for the `add_overlay` method but not for `add_contours`?

### Using add_contours (not possible to add colorbar)
![csz_neg_contour](https://user-images.githubusercontent.com/48254290/65251717-204a5e00-daf8-11e9-909d-9cf134b639ec.png)

### Using add_overlay (possible to add colorbar)
![csz_neg_overlay](https://user-images.githubusercontent.com/48254290/65251617-ec6f3880-daf7-11e9-92bd-e7868779e82c.png)

**Comment by jeromedockes** [comment]
I don't think there is a particular reason. the use case is less frequent, as I think people will tend to use a colormap (as in your second figure) to show levels, and contors to show atlas regions, e.g.  with `plot_prob_atlas`. In this case colors do not represent a scale and some kind of discrete legend could be more adapted.

Still, I think a PR adding the possibility to have a colorbar for contours would be accepted, especially if you can find a use case in the example gallery. In the meanwhile, note that this is very easy to do with matplotlib outside of nilearn

**Comment by GillesSchneider** [comment]
Hello.

@jeromedockes brought to light that a discrete legend could be more adapted in some cases. I agree with him. 
A discrete legend used besides the color bar (before adding contours) improves the overall visual interpretation of the plot. 

In the image below, the legend contains the levels. If <code>levels=[vmin, vmax]</code> is given as an argument of <code>add_contours</code>, the legend will contain an entry <code>[color] vmin < x <= vmax</code>. In addition, it will contain <code>[color] x = v</code> if <code>levels=[v]</code> is given as an argument of the method.

Please note that the legend has not been optimized yet (size, etc.). I added a single argument <code>show_level</code> to the method <code>add_contours</code> either to add or not the level to the legend. 

If it is OK for a PR, I will continue to work on it. What do you think?

Here is the code that generates the image below:

// Based on demo glass brain plotting with contours and with fillings
// Setting custom thresholds using parameter 'levels'
// If show_level is True, add level to legend
// Default=True
motor_images = fetch_neurovault_motor_task()
stat_img = motor_images.images[0]

display = plot_glass_brain(None, colorbar=False, title='plot_stat_map')

// Same colormap 
cmap = 'YlOrRd'

display.add_contours(stat_img, filled=False, levels = [1, 2], cmap=cmap)
display.add_contours(stat_img, filled=True, levels = [2, 3], colors='r')
display.add_contours(stat_img, show_level=False, filled=True, levels = [3, 6], cmap=cmap)
display.add_contours(stat_img, show_level=True, filled=True, levels = [7, np.inf], cmap=cmap)

show()

![img3](https://user-images.githubusercontent.com/57995282/118156923-ea05d480-b3e7-11eb-8875-b34deee47219.png)

**Comment by bthirion** [comment]
I like the result. But this is something else than  #2136, right ?

**Comment by GillesSchneider** [comment]
Yes, I came up with this working on a color bar for <code>add_contours</code>. Maybe I should open another issue for that. I have something for #2136, but it is not done yet.

**Comment by bthirion** [comment]
Yes, i would suggest opening another issue. But I quite like the idea !

**Comment by jeromedockes** [comment]
It would also be nice if this function returned the `QuadContourSet` created by matplotlib rather than `None` so users can easily create the colorbars or legends that they want

**Comment by NicolasGensollen** [comment]
Hi @GillesSchneider 
I also think this would be a nice addition! 
Feel free to open a dedicated issue and a PR.

**Comment by quentden** [comment]
Hi @Remi-Gau,

We have a student project with my school and I would like to know if it is possible to help on this issue ?

If yes, is it possible to know the exact place of the functions "add_contours" and "add_overlay" (I only found them in the file with the class BaseSlicer but not OrthoSlicer) ?

And also if you know with which file we can test to see if we get the expected plotting ?

Sincerely
Quentin
